views/js/runner/provider/qti.js

```diff
@@ -430,20 +390,12 @@ define([
 
                     self.disableItem(context.itemUri);
 
-                    // submit the response even if empty
-                    submit(true)
-                        .then(updateStats)
-                        .then(function() {
-                            self.trigger('alert.timeout', __('Time limit reached, this part of the test has ended.'), function() {
-                                computeNext('timeout', {
-                                    scope: scope,
-                                    ref: ref
-                                });
-                            });
-                        })
-                        .catch(function(err){
-                            self.trigger('error', err);
-                        });
+                    self.trigger('alert.timeout', __('Time limit reached, this part of the test has ended.'), function() {
+                        computeNext('timeout', _.merge(getItemResults(), {
+                            scope: scope,
+                            ref: ref
+                        }));
```

Comment:
just pushed a fixed were the timeout request is pushed before displaying the message